app/controllers/act/participants_controller.rb

```diff
@@ -0,0 +1,369 @@
+  #
+  # *Provides*
+  # @participant:: nový účastnický účet
+  # new_complet:: že to opravdu chceme i s tím uživatelským účtem
+  #
+  def new_complet
+    return redirect_to act_participant_path(current_user.participant) if Act::Participant::has_participant?(current_user)
+
+    @new_complet = true
+    @participant = Act::Participant.new()
+    render :new
+  end
+
+  ##
+  #  POST /act/participants/create
+  #
+  # Vytvoří nový účastnický účet
+  #
+  # *Params*
+  # participant_params[]:: parametry upravovaného účastnického účtu
+  #
+  # *Redirect* @participant
+  def create
+    if Act::Participant::has_participant?(current_user) && (! can? :create_other, Act::Participant)
+      @msg = "Ty už svůj uživateský účet máš."
+      return render :not_allowed
+    end
+
+    if params[:participant][:user_id] != current_user.id.to_s && (! can? :create_other, Act::Participant)
+      @msg = "Nemáš práva na vytváření uživatelských účtů ostatním užvatelům."
+      return render :not_allowed
+    end
+
+    @participant = Act::Participant.new(participant_params)
+    @user_id = current_user.id
+
+    agree = ! params[:souhlasim].nil?
+    @participant.errors.add(:souhlasim, 'Je nutno souhlasit s podmínkami') if ! agree
+
+    if @participant.errors.count == 0 && @participant.save(participant_params)
+      redirect_to @participant
+    else
+      add_alert "Pozor: ve formuláři jsou chyby"
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  ##
+  #  POST /act/participants/create
+  #
+  # Vytvoří nový účastnický účet s tím, že se současně vytvoří i uživatelský účet
+  #
+  # *Params*
+  # participant_params[]:: parametry upravovaného účastnického účtu
+  #
+  # *Redirect* :create_tnx
+  def create_complet
+    if Act::Participant::has_participant?(current_user) && (! can? :create_other, Act::Participant)
+      @msg = "Ty už svůj uživateský účet máš."
+      return render :not_allowed
+    end
+
+    @participant = Act::Participant.new(participant_params)
+
+    agree = ! params[:souhlasim].nil?
+    @participant.errors.add(:souhlasim, 'Je nutno souhlasit s podmínkami') if ! agree
+
+    if @participant.errors.count != 0 || @participant.invalid?
+      add_alert "Pozor: ve formuláři jsou chyby"
+      return render :new, status: :unprocessable_entity
+    end
+
+    send_first = true
+    user = User.find_by_email @participant.email.downcase
+    if !user
+      # create user by email
+      # Are e-mail addresses case sensitive?
+      #
+      # Yes, according to RFC 2821, the local mailbox (the portion before @)
+      # is considered case-sensitive. However, typically e-mail addresses are
+      # not case-sensitive because of the difficulties and confusion it would
+      # cause between users, the server, and the administrator.
+      user =  User.new(email: @participant.email.downcase, name: @participant.name, last_name: @participant.last_name, confirmation_sent_at: Time.now,  roles: [:user])
```

Comment:
No tak tohle se mi teda moc nelibi.  Uzivatelsky ucet by mel byt overeny.

Reply:
Která ověřovací fáze tam chybí? To, že se pošle mail, kde musí uživatel kliknout na tlačítko a pak někde zadat heslo se přece stane skrz následující dvě řádky (otestováno na tep-devu):
`user.confirm`
`user.send_first_login_instructions(false)`